### PR TITLE
Feed selection

### PR DESCRIPTION
**Pull request by rickramirr**
#### Code review checklist [Code Review Rubric](https://github.com/wizeline/iOS-apprenticeship-program-2020/blob/main/code-review-rubric.md)

1. DRY Principle (Don’t repeat yourself)

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

2. Yagni Principle (You Aren't Gonna Need It)

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

3. KISS Principle (Keep It Simple, St...d)

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

4. Remove unused code

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

5. Hard-coded values

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

6. Language style guide & coding conventions
 
 - [Wizeline Style Guide](https://github.com/wizeline/mobile-documentation/blob/master/guidelines/swift-style-guide.md) 
 - [RayWenderlich Style Guide](https://github.com/raywenderlich/swift-style-guide)

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

7. Architecture / Design Pattern

    - [x] MVC  
    - [ ] MVP 
    - [ ] MVVM 
    - [ ] VIPER 

8. Architecture Implementation

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

9. SOLID

- Single Responsibility Principle 

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Open Closed Principle

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Liskov substitutability principle

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Interface segregation

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Dependency Inversion principle

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

10. Testing

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [x] Developing

11. Force unwrapping !

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [x] Developing

12. Mutable vs immutable property (Right use of `let`  and `var`)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [x] Developing

13. Retain cycles (Capture list)

    - [x] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

14. Code Documentation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [x] Developing

### CODE REVIEW

File: TheMovieDb/Common/LoadingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        let spinner = UIActivityIndicatorView()
+        spinner.translatesAutoresizingMaskIntoConstraints = false
+        spinner.startAnimating()
+        view.addSubview(spinner)
+        
+        NSLayoutConstraint.activate([
+            spinner.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
```
Comment (Carlos-Barron) [inline TheMovieDb/Common/LoadingViewController.swift:22]: Let's keep clean the viewDidLoad. You can make functions to this responsibilities.

File: TheMovieDb/Common/LoadingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+//
+//  LoadingViewController.swift
+//  TheMovieDb
+//
+//  Created by Ricardo Ramirez on 02/11/21.
+//
+
+import UIKit
+
+class LoadingViewController: UIViewController {
```
Comment (Carlos-Barron) [inline TheMovieDb/Common/LoadingViewController.swift:10]: Are you going to inherit ?

File: TheMovieDb/Modules/Home/HomeView.swift [hunk 3]
```diff
@@ -7,10 +7,34 @@
...
+private extension FeedTypes {
+    var feedTitle: String {
+        switch self {
+        case .trending: return "Trending"
+        case .nowPlaying: return "Now Playing"
+        case .popular: return "Popular"
+        case .topRated: return "Top Rated"
+        case .upcoming: return "Upcoming"
+        }
+    }
```
Comment (Carlos-Barron) [inline TheMovieDb/Modules/Home/HomeView.swift:27]: Next session we'll discuss localizable files but for the moment it's OK

File: TheMovieDb/Modules/Home/HomeView.swift [hunk 4]
```diff
@@ -7,10 +7,34 @@
 
 import UIKit
 
+enum FeedTypes: CaseIterable {
```
Comment (Carlos-Barron) [inline TheMovieDb/Modules/Home/HomeView.swift:10]: What do you want to perform with CaseIterable?

**Review by Carlos-Barron — CHANGES_REQUESTED** [review]
Check the comments

File: TheMovieDb/Common/LoadingViewController.swift (hunk 2 above)
Comment (rickramirr) [reply TheMovieDb/Common/LoadingViewController.swift:10]: Hahah always forget about it

File: TheMovieDb/Modules/Home/HomeView.swift (hunk 4 above)
Comment (rickramirr) [reply TheMovieDb/Modules/Home/HomeView.swift:10]: I need it to get the types array and use it on the collection delegate and data source

`
func collectionView(
        _ collectionView: UICollectionView,
        numberOfItemsInSection section: Int
    ) -> Int {
        return FeedTypes.allCases.count
    }
`

File: TheMovieDb/Common/LoadingViewController.swift (hunk 1 above)
Comment (rickramirr) [reply TheMovieDb/Common/LoadingViewController.swift]: Solved

File: TheMovieDb/Common/LoadingViewController.swift (hunk 2 above)
Comment (rickramirr) [reply TheMovieDb/Common/LoadingViewController.swift]: Solved

File: TheMovieDb/Modules/Home/HomeView.swift (hunk 3 above)
Comment (rickramirr) [reply TheMovieDb/Modules/Home/HomeView.swift:27]: Oks

**Comment by rickramirr** [comment]
Submitted some fixes :)

**Review by Carlos-Barron — APPROVED** [review]

**Pull request merged**